lib/services/profiles-api-service.js

```diff
@@ -375,33 +372,44 @@ function profileApiServiceFactory(
 
     ProfileApiService.prototype.getProfiles = function(req, query, res) {
         var self = this;
-        return this.setLookup(req, res)
-            .then(function() {
-                var macs = query.mac || query.macs;
-                if (macs) {
-                    var macAddresses = self.getMacs(macs);
-                    var options = {
-                        type: 'compute'
-                    };
+        var ipAddress = res.locals.ipAddress;
+        return self.getRequestMacAddress(query, ipAddress)
+        .then(function(macAddress) {
+            if(macAddress) {
+                res.locals.macAddress = macAddress;
+                var proxyIp = req.get(Constants.HttpHeaders.ApiProxyIp);
+                var proxyPort = req.get(Constants.HttpHeaders.ApiProxyPort);
+                return self.setLookup(ipAddress, macAddress, proxyIp, proxyPort);
+            }
+        })
+        .then(function() {
+            var macs = query.mac || query.macs;
+            if (macs) {
+                var macAddresses = self.getMacs(macs);
+                var options = {
+                    type: 'compute'
+                };
 
-                    return self.getNode(macAddresses, options)
-                        .then(function (node) {
-                            return self.getProfileFromTaskOrNode(node, 'compute')
-                                .then(function (render) {
-                                    return(render);
+                return self.getNode(macAddresses, options)
+                    .then(function (node) {
+                        return self.getProfileFromTaskOrNode(node, 'compute')
+                            .then(function (render) {
+                                return _.defaults(render, {
+                                    ignoreLookup: res.locals.macAddress ? true : false
```

Comment:
it will leads to macAddress render missing,

Reply:
Here I use res.locals.macAddress to render macAddress, https://github.com/RackHD/on-http/pull/764/files#diff-65e9fcdeffbfaa4a0d0b2de5bd283da6R294